elastic/kibana
# [Console] Handle the case where saving to history may throw an error

## Summary

We should still show the request result even if localStorage is full (or any other save error occurs).

Fix https://github.com/elastic/kibana/issues/62203

### How to test

Run the following script in browser console:

```javascript
const prefix = 'THIS_IS_GARBAGE_DATA_'

function fillLocalStorage() {
  const data = 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF'
  let idx = 0
  while(true) {
    localStorage[prefix + ++idx] = data
  }
}

function undoFillLocalStorage() {
  Object.keys(localStorage).forEach(k => {
    if (k.startsWith(prefix)) {
      localStorage.removeItem(k)
    }
  })
}
```

Then run `fillLocalStorage()` and wait for the localStorage quota error to appear.

Run a request in Console and see the error notification appear, the request result should still appear in the output pane.

Run `undoFillLocalStorage()`. Then run a Console request again, this time there should be no error about localStorage quota.

<details>
<summary>Gif of request with history error</summary>

![request-with-history-error](https://user-images.githubusercontent.com/8155004/78355182-96527600-75ad-11ea-8748-d5146bd9f798.gif)

</details>

### Release Note

We fixed an issue where Console would not render the request output if localStorage quota was reached.
